catlearn/learning_curve/learning_curve.py

```diff
@@ -1,11 +1,95 @@
 """Generate the learning curve."""
 import numpy as np
 
+import time
+import multiprocessing
+from tqdm import trange, tqdm
+
 from .data_process import data_process
 from .placeholder import placeholder
 from catlearn.preprocess.scaling import target_normalize
 
 
+class LearningCurve(object):
+    """Learning curve class. Test a model while varying
+    the density of the training data."""
+    def __init__(self, nprocs=1):
+        """Initialize the class.
+
+        Parameters
+        ----------
+        nprocs : int
+            Number of processers used in parallel implementation. Default is 1
+            e.g. serial.
+        """
+        self.nprocs = nprocs
+
+    def run(self, predict, train, target, test, test_target,
+            step=1, min_data=2):
+        """Evaluate custom metrics versus training data size.
+
+        Parameters
+        ----------
+        predict : object
```

Comment:
Instead of predict, can we use classifier/algorithm/machine. Since it will both fit the training data and predict the testing data, this name is slightly misleading. When I read it, I assumed you passed a already trained model, but in the function it will change the train/test ratio and rerun the whole process again for each ratio.

Reply:
Indeed it is a very flexible function. It could read a model from a file, update data and predict, or train and predict. What about `model` or `function`? I'll improve the docstring.